dipy/tracking/local/localtrack.pyx

```diff
@@ -208,3 +210,254 @@ cdef int _local_tracker(
         # maximum length of streamline has been reached, return everything
         i = streamline.shape[0]
     return i
+
+
+def pft_tracker(
+        DirectionGetter dg,
+        ConstrainedTissueClassifier tc,
+        np.float_t[:] seed,
+        np.float_t[:] first_step,
+        np.float_t[:] voxel_size,
+        np.float_t[:,:] streamline,
+        np.float_t[:,:] directions,
+        double step_size,
+        int pft_nbr_back_steps,
+        int pft_max_steps,
+        int pft_max_trial,
+        int particle_count,
+        np.float_t[:,:,:,:] particle_paths,
+        np.float_t[:,:,:,:] particle_dirs,
+        np.float_t[:,:] particle_weights,
+        np.int_t[:,:,:]  particle_states):
+    cdef:
+        int i
+        TissueClass tissue_class
+
+    if (seed.shape[0] != 3 or first_step.shape[0] != 3 or
+            voxel_size.shape[0] != 3 or streamline.shape[1] != 3):
+        raise ValueError()
+
+    i = _pft_tracker(dg, tc, seed, first_step, voxel_size, streamline,
+                     directions, step_size, &tissue_class, pft_nbr_back_steps,
+                     pft_max_steps, pft_max_trial, particle_count,
+                     particle_paths, particle_dirs, particle_weights,
+                     particle_states)
+    return i, tissue_class
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+cdef _pft_tracker(DirectionGetter dg,
+                  ConstrainedTissueClassifier tc,
+                  np.float_t[:] seed,
+                  np.float_t[:] first_step,
+                  np.float_t[:] voxel_size,
+                  np.float_t[:,:] streamline,
+                  np.float_t[:,:] directions,
+                  double step_size,
+                  TissueClass* tissue_class,
+                  int pft_nbr_back_steps,
+                  int pft_max_steps,
+                  int pft_max_trial,
+                  int particle_count,
+                  np.float_t[:,:,:,:] particle_paths,
+                  np.float_t[:,:,:,:] particle_dirs,
+                  np.float_t[:,:] particle_weights,
+                  np.int_t[:,:,:]  particle_states):
+
+    if (seed.shape[0] != 3 or first_step.shape[0] != 3 or
+            voxel_size.shape[0] != 3 or streamline.shape[1] != 3):
+        raise ValueError()
+    cdef:
+        int i, pft_trial, pft_streamline_i, pft_nbr_steps
+        double point[3], dir[3], vs[3], voxdir[3]
+
+        void (*step)(double * , double*, double) nogil
+    pft_trial = 0
+
+    for j in range(3):
+        streamline[0, j] = point[j] = seed[j]
+        dir[j] = first_step[j]
+        vs[j] = voxel_size[j]
+    copypoint(dir, &directions[0, 0])
+
+    tissue_class[0] = TRACKPOINT
+    i = 0
+    while i < streamline.shape[0] - 1:
+        if dg.get_direction(point, dir):
```

Comment:
`get_direction_c` is the fast call (from cython to cython instead of cython to python)